source/draw/RasterMonochrome.ooc

```diff
@@ -49,21 +49,63 @@ RasterMonochrome: class extends RasterPacked {
 			}
 	}
 	resizeTo: override func (size: IntSize2D) -> This {
+		this resizeTo(size, TransformMethod Smooth) as This
+	}
+	resizeTo: override func ~withMethod (size: IntSize2D, method: TransformMethod) -> This {
 		result: This
 		if (this size == size)
 			result = this copy()
 		else {
 			result = This new(size)
-			for (row in 0 .. size height) {
-				srcRow := (this size height * row) / size height
-				for (column in 0 .. size width) {
-					srcColumn := (this size width * column) / size width
-					result[column, row] = this[srcColumn, srcRow]
-				}
+			match (method) {
+				case TransformMethod Fast => This _resizeNearestNeighbour(this, result)
+				case TransformMethod Smooth => This _resizeBilinear(this, result)
+				case =>
```

Comment:
Someone could cast `1337 as TransformMethod`, I guess. We *could* do it this way and return an empty image when someone uses an invalid `TransformMethod`, of course. We could also throw an exception. I'm just asking what you think.